osbenchmark/builder/launchers/local_process_launcher.py

```diff
@@ -0,0 +1,187 @@
+
+        java_major_version, java_home = java_resolver.java_home(node_configuration.provision_config_instance_runtime_jdks,
+                                                                self.pci.variables["system"]["runtime"]["jdk"],
+                                                                node_configuration.provision_config_instance_provides_bundled_jdk)
+        self.logger.info("Java major version: %s", java_major_version)
+        self.logger.info("Java home: %s", java_home)
+        self.logger.info("Starting node [%s].", node_name)
+
+        telemetry = self._prepare_telemetry(node_configuration, node_count_on_host, java_major_version)
+        env = self._prepare_env(node_name, java_home, telemetry)
+        telemetry.on_pre_node_start(node_name)
+
+        node_pid = self._start_process(host, binary_path, env)
+        self.logger.info("Successfully started node [%s] with PID [%s].", node_name, node_pid)
+        node = cluster.Node(node_pid, binary_path, host_name, node_name, telemetry)
+
+        self.logger.info("Attaching telemetry devices to node [%s].", node_name)
+        telemetry.attach_to_node(node)
+
+        return node
+
+    def _prepare_telemetry(self, node_configuration, node_count_on_host, java_major_version):
+        data_paths = node_configuration.data_paths
+        node_telemetry_dir = os.path.join(node_configuration.node_root_path, "telemetry")
+
+        enabled_devices = self.pci.variables["telemetry"]["devices"]
+        telemetry_params = self.pci.variables["telemetry"]["params"]
+
+        node_telemetry = [
+            telemetry.FlightRecorder(telemetry_params, node_telemetry_dir, java_major_version),
+            telemetry.JitCompiler(node_telemetry_dir),
+            telemetry.Gc(telemetry_params, node_telemetry_dir, java_major_version),
+            telemetry.Heapdump(node_telemetry_dir),
+            telemetry.DiskIo(node_count_on_host),
+            telemetry.IndexSize(data_paths),
+            telemetry.StartupTime(),
+        ]
+
+        return telemetry.Telemetry(enabled_devices, devices=node_telemetry)
+
+    def _prepare_env(self, node_name, java_home, telemetry):
+        env = {k: v for k, v in os.environ.items() if k in opts.csv_to_list(self.pci.variables["system"]["env"]["passenv"])}
+        if java_home:
+            self._set_env(env, "PATH", os.path.join(java_home, "bin"), separator=os.pathsep, prepend=True)
+            # This property is the higher priority starting in ES 7.12.0, and is the only supported java home in >=8.0
+            env["OPENSEARCH_JAVA_HOME"] = java_home
+            # TODO remove this when ES <8.0 becomes unsupported by Benchmark
+            env["JAVA_HOME"] = java_home
+            self.logger.info("JAVA HOME: %s", env["JAVA_HOME"])
+        if not env.get("OPENSEARCH_JAVA_OPTS"):
+            env["OPENSEARCH_JAVA_OPTS"] = "-XX:+ExitOnOutOfMemoryError"
+
+        # we just blindly trust telemetry here...
```

Comment:
Is this a note or will this be a TODO to refactor this portion so that we don't blindly trust telemetry in the future?

Reply:
This is an inherited comment. I don't think it's a TODO so much as calling out that the telemetry devices can set environment variables that may break the launching of OpenSearch. Essentially loosely passing the burden of validating the env vars that a telemetry device specifies to the telemetry device itself.

A better solution would probably be an allowlist, but I don't have enough familiarity with the scope of telemetry environment variables to propose one.